.github/workflows/e2e.yaml

```diff
@@ -11,6 +11,14 @@ jobs:
     name: Build & Run E2E Images
     runs-on: [self-hosted, linux, X64, jammy, large]
     steps:
+      -
+        name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          # We run into rate limiting issues if we don't authenticate
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
       - name: Check out repo
         uses: actions/checkout@v4
       - name: Install requirements
```

Comment:
Just a note: make sure to avoid too high coupling between the workflow itself and the setting up of the CI (e.g. runners). It works now, but the danger would be that it becomes very brittle and one of those things that people are afraid to touch.

Reply:
You mean the Github login should not be a separate step? Where would you put this instead?